llvm/lib/Object/MachOObjectFile.cpp

```diff
@@ -5193,7 +5193,7 @@ MachOObjectFile::getDyldChainedFixupTargets() const {
   const char *Symbols = Contents + Header.symbols_offset;
   const char *SymbolsEnd = Contents + DyldChainedFixups.datasize;
 
-  if (ImportsEnd > Symbols)
+  if (ImportsEnd > SymbolsEnd)
     return malformedError("bad chained fixups: imports end " +
                           Twine(ImportsEndOffset) + " extends past end " +
                           Twine(DyldChainedFixups.datasize));
```

Comment:
If I look at line 5201, form what I understand, a `{DyldChainedFixups.dataoff, DyldChainedFixups.datasize }` span first contains a span of Imports, and following it a span of Symbols (in valid files). Is that correct?

If so, the `if (ImportsEnd > Symbols)` check in line 5201 makes sure that the Imports span is in fact in front of Symbols, and even after this patch this if here can never be true (?) Or did I get this wrong?

Reply:
> If I look at line 5201, form what I understand, a {DyldChainedFixups.dataoff, DyldChainedFixups.datasize } span first contains a span of Imports, and following it a span of Symbols (in valid files). Is that correct?

Correct.

> If so, the if (ImportsEnd > Symbols) check in line 5201 makes sure that the Imports span is in fact in front of Symbols, and even after this patch this if here can never be true (?) Or did I get this wrong?

this is checked *before* `ImportsEnd > Symbols`, so if `ImportsEnd` extends past the whole section (i.e. `SymbolsEnd`), this message will be printed instead of the one belonging to line 5201. But I agree that there is not much point in making this distinction. Let's just remove this one.